### PR TITLE
OSDOCS-11942 adding Azure CAPI release note

### PR DESCRIPTION
**Pull request by bscott-rh**
Version(s):
4.17

Issue:
https://issues.redhat.com/browse/OSDOCS-11942

Link to docs preview:
https://81768--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-17-release-notes.html#ocp-4-17-installation-and-update-azure-capi_release-notes

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by bscott-rh** [comment]
@jinyunma PTAL at this release note announcing Azure CAPI for 4.17. Thank you

File: release_notes/ocp-4-17-release-notes.adoc [hunk 1]
```diff
@@ -55,6 +55,10 @@ This release adds improvements related to the following components and concepts:
 [id="ocp-4-17-installation-and-update_{context}"]
 === Installation and update
 
+[id="ocp-4-17-installation-and-update-azure-capi_{context}"]
+==== Cluster API replaces Terraform for {azure-first} installations
+In {product-title} 4.17, the installation program uses Cluster API instead of Terraform to provision cluster infrastructure during installations on {azure-short}.
```
Comment (jinyunma) [inline release_notes/ocp-4-17-release-notes.adoc:60]: After switching to use Cluster API, more permissions are required if user uses service principal with minimal permissions to install cluster, it's better to highlight here.

Missed permissions are tacked via https://issues.redhat.com/browse/OCPBUGS-37701.

Comment (bscott-rh) [reply release_notes/ocp-4-17-release-notes.adoc:64]: Thank you @jinyunma I have updated this PR with a note about the permissions, and added the permissions to the main documentation in a separate PR: https://github.com/openshift/openshift-docs/pull/82095 Please review, thanks

Comment (jinyunma) [reply release_notes/ocp-4-17-release-notes.adoc:64]: Thanks for the update, the change looks good to me.

**Review by cbippley — COMMENTED** [review]
LGTM!

**Pull request merged**